rotkehlchen/tests/api/test_bitcoin.py

```diff
@@ -320,22 +327,23 @@ def test_add_delete_xpub_multiple_chains(rotkehlchen_api_server):
     # Check that addresses that have balances on both bitcoin and bitcoin cash are stored properly
     with rotki.data.db.conn.read_ctx() as cursor:
         mutual_in_chains_aggregator = 0
-        for btc_addr in rotki.chains_aggregator.accounts.btc:
+        btc_addesses: tuple[BTCAddress, ...] = rotki.chains_aggregator.accounts.btc
+        for btc_addr in btc_addesses:
             if btc_addr in rotki.chains_aggregator.accounts.bch:
                 mutual_in_chains_aggregator += 1
         # get mutual by querying BCH accounts from known BTC accounts
         # this checks that bitcoin mappings in the db are not broken
         mutual_in_db_1 = len(rotki.data.db.get_addresses_to_xpub_mapping(
             cursor=cursor,
             blockchain=SupportedBlockchain.BITCOIN,
-            addresses=rotki.chains_aggregator.accounts.bch,
+            addresses=list(rotki.chains_aggregator.accounts.bch),
         ))
         # get mutual by querying BTC accounts from known BCH accounts
         # this checks that bitcoin cash mappings in the db are not broken
         mutual_in_db_2 = len(rotki.data.db.get_addresses_to_xpub_mapping(
             cursor=cursor,
             blockchain=SupportedBlockchain.BITCOIN_CASH,
-            addresses=rotki.chains_aggregator.accounts.btc,
+            addresses=list(rotki.chains_aggregator.accounts.btc),
```

Comment:
why is the `list()` needed? Does that not mean that the original type requirement is off? Why not make `get_addresses_to_xpub_mapping` accept a  `Sequence[BTCAddress]` instead?